subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/transform/ArtifactTransformBuildOperationIntegrationTest.groovy

```diff
@@ -1246,4 +1262,15 @@ class ArtifactTransformBuildOperationIntegrationTest extends AbstractIntegration
             subjectName == expectedDetails.subjectName
         }
     }
+
+    void checkExecuteTransformWorkOperations(BuildOperationRecord executePlannedTransformStep, int expectOperationsCount) {
+        def executeWorkOps = buildOperations.children(executePlannedTransformStep, ExecuteWorkBuildOperationType)
+        assert executeWorkOps.size() == expectOperationsCount
+        executeWorkOps.every {
+            verifyAll(it) {
+                details.workType == "TRANSFORM"
+                result.skipMessage == null
```

Comment:
❓ I suppose we should have tests for the different outcomes, right?

Reply:
Yes, we need to add them, the existing tests don't seem to cover cacheability and incremental builds.

Reply:
I ended up adding one integration test that checks for both `UP-TO-DATE` and `FROM-CACHE`